### PR TITLE
Optimize the connect code

### PR DESCRIPTION
**Pull request by BooniesFX**
1. rm legacy node when reconnect
2. add back off time when reconnect
3. Create a new routine to check network connection

Signed-off-by: Xiang Fu <fuxiang@onchain.com>

### CODE REVIEW

File: net/node/infoUpdate.go [hunk 1]
```diff
@@ -122,58 +122,55 @@ func getNodeAddr(n *node) NodeAddr {
...
+			if tn.conn != nil {
+				tn.CloseConn()
+			}
+		} else {
+			//add others to tmp node map
+			n.RemoveFromRetryList(nodeAddr)
+			neibornodes[tn.GetID()] = tn
+		}
+	}
+	n.nbrNodes.List = neibornodes
```
Comment (laizy) [inline net/node/infoUpdate.go]: use write Lock instead since you are modifying the internal data.

Comment (laizy) [inline net/node/infoUpdate.go]: notice that it defines two struct with a RWMutex.
```golang
type RetryConnAddrs struct {
	sync.RWMutex
	RetryAddrs map[string]int
 }
type ConnectingNodes struct {
	sync.RWMutex
	ConnectingAddrs []string
}
```
and move the addr in one list to another not in an atomic step ,  which will cause concurrent consistency problem。maybe need managed with the same Lock like the following struct :
```golang
type connAddrs struct {
	sync.RWMutex
	RetryAddrs map[string]int
        ConnectingAddrs []string
 }
```

Comment (BooniesFX) [inline net/node/infoUpdate.go]: These two lists are independent, one ip address can both in retry list and connecting list. So now use two struct. 
And RLock replaced.

Comment (laizy) [inline net/node/infoUpdate.go]: :ok_hand:

**Pull request merged**